packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -42,26 +49,47 @@ library LibProposing {
         internal
         returns (TaikoData.BlockMetadata memory meta)
     {
-        uint8 cacheTxListInfo = _validateBlock({
-            state: state,
-            config: config,
-            input: input,
-            txList: txList
-        });
-
-        if (cacheTxListInfo != 0) {
-            state.txListInfo[input.txListHash] = TaikoData.TxListInfo({
-                validSince: uint64(block.timestamp),
-                size: uint24(txList.length)
+        // Try to select a prover first to revert as earlier as possible
+        (address assignedProver, uint32 rewardPerGas) = IProverPool(
+            resolver.resolve("prover_pool", false)
+        ).assignProver(state.numBlocks, state.feePerGas);
```

Comment:
To handle the max block speed, the prover has to set high capacity, which means large computation power should be prepared. And then, the cost might be high, so the prover needs to be able to adjust the computation power dynamically to save cost.

Reply:
anyway, it's not big issue, because the required staking token amount for each prover is bigger than 128*10000*10e-8 TTKO.